File: Remote_Labs/Test_Equipment/FCPM-6000RC Freq Ctr Power Meter/MCL_PowerMeter_Examples/mcl_pm64.dll_Examples/VB.NET/WindowsApplication1/WindowsApplication1/Form1.vb
Public Class Form1
    Dim pm1 As New mcl_pm64.usb_pm, Status As Short
    Dim StopRun As Boolean
    
    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click

        Dim SN As String = "" 'if more then 1 sensor connected to the computer than the Serial Number of the sensor should provide
        Dim ReadResult As Single
        Status = pm1.Open_Sensor(SN)
        StopRun = False
        While Not StopRun
            pm1.Freq = Val(TextBox1.Text)   ' Set the Frequency cal factor in MHz
            ReadResult = pm1.ReadPower() ' read the power in dbm
            Label3.Text = Format$(ReadResult, "0.00")
            Application.DoEvents()
        End While
        ' Recommanded to close object when terminate the program
        'pm1.Close_Sensor()
    End Sub

    Private Sub Button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button2.Click
        StopRun = True
    End Sub

    Private Sub Form1_FormClosing(ByVal sender As Object, ByVal e As System.Windows.Forms.FormClosingEventArgs) Handles Me.FormClosing
        pm1.Close_Sensor()
    End Sub

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load

    End Sub
End Class

File: Remote_Labs/Test_Equipment/FCPM-6000RC Freq Ctr Power Meter/MCL_PowerMeter_Examples/mcl_pm64.dll_Examples/C#.NET/WindowsApplication1/WindowsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {
        bool StopRun;
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // mcl_pm64.dll should be loaded as a reference file to the project project
            double ReadResult;
            string FreqSTR = "";
            mcl_pm64.usb_pm pm1 = new mcl_pm64.usb_pm();
            short Status = 0;
            string pm_SN = ""; //if more then 1 sensor connected to the computer than the Serial Number of the sensor should provide
            Status = pm1.Open_Sensor(ref(pm_SN));
            StopRun = false;
            while (!StopRun)
            {
            FreqSTR = textBox1.Text;
            if (FreqSTR == "") FreqSTR = "10";   
            pm1.Freq = Convert.ToDouble(FreqSTR);      // Set the Frequency cal factor in MHz
            ReadResult = pm1.ReadPower(); // read the power in dbm
            label3.Text = string.Format("{0:0.00}",  ReadResult)  ; 
            Application.DoEvents();
            }
            pm1.Close_Sensor();  
        }

        private void button2_Click(object sender, EventArgs e)
        {
            StopRun = true;
        }

        
    }
}